restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/GeolocationEndpoint.java

```diff
@@ -0,0 +1,732 @@
+                            data.putSingle("LocationAreaCode", token);
+                        }
+                        if (item.contains("cellid")) {
+                            String token = item.substring(item.lastIndexOf("=") + 1);
+                            data.putSingle("CellId", token);
+                        }
+                        if (item.contains("aol")) {
+                            String token = item.substring(item.lastIndexOf("=") + 1);
+                            data.putSingle("LocationAge", token);
+                        }
+                        if (item.contains("vlrNumber")) {
+                            String token = item.substring(item.lastIndexOf("=") + 1);
+                            data.putSingle("NetworkEntityAddress", token);
+                        }
+                        if (item.contains("latitude")) {
+                            String token = item.substring(item.lastIndexOf("=") + 1);
+                            data.putSingle("DeviceLatitude", token);
+                        }
+                        if (item.contains("longitude")) {
+                            String token = item.substring(item.lastIndexOf("=") + 1);
+                            data.putSingle("DeviceLongitude", token);
+                        }
+                        if (item.contains("civicAddress")) {
+                            String token = item.substring(item.lastIndexOf("=") + 1);
+                            data.putSingle("FormattedAddress", token);
+                        }
+                    }
+                }
+                if (gmlcURI != null && gmlcResponse != null) {
+                    // For debugging/logging purposes only
+                    logger.info("Geolocation data of " + targetMSISDN + " retrieved from GMCL at: " + gmlcURI);
+                    logger.info("MCC (Mobile Country Code) = " + getInteger("MobileCountryCode", data));
+                    logger.info("MNC (Mobile Network Code) = " + data.getFirst("MobileNetworkCode"));
+                    logger.info("LAC (Location Area Code) = " + data.getFirst("LocationAreaCode"));
+                    logger.info("CI (Cell ID) = " + data.getFirst("CellId"));
+                    logger.info("AOL (Age of Location) = " + getInteger("LocationAge", data));
+                    logger.info("NNN (Network Node Number/Address) = " + +getLong("NetworkEntityAddress", data));
+                    logger.info("Devide Latitude = " + data.getFirst("DeviceLatitude"));
+                    logger.info("Devide Longitude = " + data.getFirst("DeviceLongitude"));
+                    logger.info("Civic Address = " + data.getFirst("FormattedAddress"));
+                }
+            }
+        } catch (Exception ex) {
+            ex.printStackTrace();
+        }
+
+        Geolocation geolocation = createFrom(new Sid(accountSid), data, geolocationType);
+
+        if (geolocation.getResponseStatus() != null
```

Comment:
Sorry @gvagenas, not sure if I understood well on this one. Do you mean to better do the following?

```
if (!geolocation.getResponseStatus().equals(responseStatus.Rejected.toString()) {
    dao.addGeolocation(geolocation);
    if (APPLICATION_XML_TYPE == responseType) {
            final RestCommResponse response = new RestCommResponse(geolocation);
            return ok(xstream.toXML(response), APPLICATION_XML).build();
     } else if (APPLICATION_JSON_TYPE == responseType) {
              return ok(gson.toJson(geolocation), APPLICATION_JSON).build();
     } else {
             return null;
     }
}
```